current/integrate/subkey.md

```diff
@@ -44,212 +44,276 @@ Generate an sr25519 key by running:
+  the key pair. All other information is calculated from the seed.
+- **Public Key** (aka "Account ID") - The public half of the cryptographic key pair in hexadecimal.
+- **SS58 Address** (aka "Public Address") - An SS58-encoded address based on the public key.
 
-You can also create a vanity address, although you will not receive a mnemonic seed:
+Currently `subkey` supports the following cryptographic key pairs and signing algorithms:
+
+- `-s`, `--sr25519`: Schorr signatures on the Ristretto group
+- `-e`, `--ed25519`: ed25519
+- `-k`, `--secp256k1`: ECDSA signatures on secp256k1
+
+Note that the address for a secp256k1 key is the SS58 encoding of the hash of the public key in
+order to reduce the public key from 33 bytes to 32 bytes.
+
+You can also create a vanity address, meaning an address that contains a specified sub-string. But
+you will not receive a mnemonic phrase for this address.
 
 ```bash
 $ subkey vanity joe  --number 1
 Generating key containing pattern 'joe'
+100000 keys searched; best is 185/189 complete
+200000 keys searched; best is 187/189 complete
+300000 keys searched; best is 187/189 complete
 best: 189 == top: 189
-Secret Key URI `0xc645213712dc218e851ac0a47e987ee3f8e1cbbbad85522dbbf86e51963de434` is account:
-  Public key (hex): 0xfaff3b483564c4d2dab7229e4369841d3650a4ca9c36094e9faece3839ac6e06
-  Account ID:       0xfaff3b483564c4d2dab7229e4369841d3650a4ca9c36094e9faece3839ac6e06
-  Address (SS58): 5HjoeXxUokiJwQPRucDXCgGzKwwf74q9srJcmsaFssUW2UHn
+Secret Key URI `0x380ba96e07933b89c2b3b6d3fa98b695aab99070b8982817b74044c6fa6a375b` is account:
+  Secret seed:      0x380ba96e07933b89c2b3b6d3fa98b695aab99070b8982817b74044c6fa6a375b
+  Public key (hex): 0x4a0e48c38ae880f7eee86c8a75c6391ecb2de35adf49ee30cd1631e82de1b001
+  Account ID:       0x4a0e48c38ae880f7eee86c8a75c6391ecb2de35adf49ee30cd1631e82de1b001
+  SS58 Address:     5DjoeJSa7m4EsuSKwDN1GkrMtTiWwSMGxESQ6KSNPETPaPna
 ```
 
+Notice the SS58 Address 5D**joe**JSa7m4EsuSKwDN1GkrMtTiWwSMGxESQ6KSNPETPaPna contains the string
+`joe`.
+
+`--number` is for legacy purpose, so it will generate one vanity address no matter what value is
+passed in.
+
 ## Password Protected Keys
 
-To generate a key that is password protected, use the `-p <password>` or `--password <password>` flag:
+To generate a key that is password protected, use the `-p <password>` or `--password <password>`
+flag:
 
 ```bash
-$ subkey --password "correct horse battery staple" generate
-Secret phrase `razor blouse enroll maximum lobster bacon raccoon ocean law question worry length` is account:
-  Secret seed: 0x806e324821f6b09fa0329b7da35b8056b452fdcaadeac463ff2b8db1fc020c3e
-  Public key (hex): 0xe2b410bdc959fcc7f756e6eefdb051000c80be339acd3a8dc6d941211160b266
-  Account ID:       0xe2b410bdc959fcc7f756e6eefdb051000c80be339acd3a8dc6d941211160b266
-  Address (SS58): 5HBxAo81NWBbg6DQNbTdFg7GWQDS1QmkMwNLQso1e11DfQ2Y
+$ subkey --password "pencil laptop kitchen cutter" generate
```

Comment:
I would still like to use `correct horse battery staple` as it's a [well-known](https://xkcd.com/936/) and funny password. Not critical though.

Reply:
I am trying to fix this, but then I see there are dependencies below. I used this [secret phrase, seed, password] set to demonstrate other concepts below.

So feel lazy to update them all :)